tools/resourcedocsgen/cmd/docs/registry.go

```diff
@@ -234,6 +307,115 @@ func resourceDocsFromRegistryCmd() *cobra.Command {
+	}
+
+	var metadata pkg.PackageMeta
+	if err := yaml.Unmarshal(b, &metadata); err != nil {
+		return nil, errors.Wrapf(err, "unmarshalling the metadata file %s", metadataFilePath)
+	}
+
+	return &metadata, nil
+}
+
+// ListPackageMetadata implements PackageMetadataProvider for FileSystemProvider
+func (p *FileSystemProvider) ListPackageMetadata() ([]*pkg.PackageMeta, error) {
+	registryPackagesPath := getRegistryPackagesPath(p.registryDir)
+	files, err := os.ReadDir(registryPackagesPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "reading directory %s", registryPackagesPath)
+	}
+
+	// Count YAML files to pre-allocate the slice mostly to appease the linter.
+	var metadataCount int
+	for _, file := range files {
+		if strings.HasSuffix(file.Name(), ".yaml") {
+			metadataCount++
+		}
+	}
+
+	metadataList := make([]*pkg.PackageMeta, 0, metadataCount)
+	for _, file := range files {
+		if !strings.HasSuffix(file.Name(), ".yaml") {
+			continue
+		}
+
+		metadata, err := p.GetPackageMetadata(strings.TrimSuffix(file.Name(), ".yaml"))
+		if err != nil {
+			return nil, err
+		}
+		metadataList = append(metadataList, metadata)
+	}
+
+	return metadataList, nil
+}
+
+// GetPackageMetadata implements PackageMetadataProvider for RegistryAPIProvider
+func (p *RegistryAPIProvider) GetPackageMetadata(pkgName string) (*pkg.PackageMeta, error) {
+	resp, err := http.Get(fmt.Sprintf("%s/packages?name=%s", p.apiURL, pkgName))
+	if err != nil {
+		return nil, errors.Wrapf(err, "fetching package metadata from API for %s", pkgName)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.Errorf("unexpected status code %d when fetching package metadata", resp.StatusCode)
+	}
+
+	var response PackageListResponse
+	if err := json.NewDecoder(resp.Body).Decode(&response); err != nil {
+		return nil, errors.Wrap(err, "decoding API response")
+	}
+
+	if len(response.Packages) == 0 {
+		return nil, errors.Errorf("no package found with name %s", pkgName)
+	}
+
+	if len(response.Packages) > 1 {
+		return nil, errors.Errorf("multiple packages found with name %s", pkgName)
+	}
+
+	return convertAPIPackageToPackageMeta(response.Packages[0])
+}
+
+// ListPackageMetadata implements PackageMetadataProvider for RegistryAPIProvider
+func (p *RegistryAPIProvider) ListPackageMetadata() ([]*pkg.PackageMeta, error) {
+	resp, err := http.Get(p.apiURL + "/packages")
+	if err != nil {
+		return nil, errors.Wrap(err, "fetching package list from API")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.Errorf("unexpected status code %d when fetching package list", resp.StatusCode)
+	}
+
+	var response PackageListResponse
+	if err := json.NewDecoder(resp.Body).Decode(&response); err != nil {
+		return nil, errors.Wrap(err, "decoding API response")
+	}
+
+	metadataList := make([]*pkg.PackageMeta, 0, len(response.Packages))
+	for _, apiPkg := range response.Packages {
```

Comment:
This is a paginated API, so this might need to call GET more then once.

Reply:
ahh good call! will adjust!